misk-redis/src/test/kotlin/misk/redis/AbstractRedisTest.kt

```diff
@@ -1289,6 +1311,196 @@ abstract class AbstractRedisTest {
     assertEquals(0, redis.zcard(key))
   }
 
+  @Test
+  fun `pipelining works - regular keys`() {
+    val suppliers = mutableListOf<Supplier<*>>()
+    redis.pipelining {
+      suppliers.addAll(
+        listOf(
+          set("test key", "test value".encodeUtf8()),
+          get("test key"),
+          setnx("keynx", "valuenx".encodeUtf8()),
+          setnx("keynx", "valuenx2".encodeUtf8()),
+          get("keynx"),
+          incr("incr"),
+          incrBy("incr", 3),
+          get("incr"),
+          del("test key")
+        )
+      )
+    }
+    assertThat(suppliers.map { it.get() }).containsExactly(
+      Unit,
+      "test value".encodeUtf8(),
+      true,
+      false,
+      "valuenx".encodeUtf8(),
+      1L,
+      4L,
+      "4".encodeUtf8(),
+      true
+    )
+  }
+
+  @Test
+  fun `pipelining works - multikey commands`() {
+    val suppliers = mutableListOf<Supplier<*>>()
+
+    redis.pipelining {
+      suppliers.addAll(
+        listOf(
+          mset(
+            "mkey1".encodeUtf8(),
+            "mval1".encodeUtf8(),
+            "mkey2".encodeUtf8(),
+            "mval2".encodeUtf8()
+          ),
+          mget("mkey1", "mkey2"),
```

Comment:
nb: in clustered mode, the underlying impl actually makes two `mset` and two `mget` calls. This keeps the slotting behaviour opaque to the caller (matching RealRedis).